src/Nest/Aggregations/AggregationContainer.cs

```diff
@@ -287,6 +290,15 @@ public void Accept(IAggregationVisitor visitor)
 		{
 			if (visitor.Scope == AggregationVisitorScope.Unknown) visitor.Scope = AggregationVisitorScope.Aggregation;
 			new AggregationWalker().Walk(this, visitor);
+		}
+
+		internal static void InjectMetadata(IAggregation aggregation)
+		{
+			if (aggregation.Meta == null)
+				aggregation.Meta = new Dictionary<string, object>();
+			if (aggregation.Meta.ContainsKey(AggregationMetadata.Key))

```

Comment:
Should a user be prevented from adding that key to metadata to begin i.e. introduce our own metadata dictionary that encapsulates this rule?